.cirrus.yml

```diff
@@ -59,6 +59,18 @@ validate_task:
         image_name: ${DEBIAN_CACHE_IMAGE_NAME}
     env:
         HOME: "/root"  # default unset, needed by golangci-lint.
+    go_modules_cache: &go_modules_cache
+        fingerprint_script: cat go.sum
+        folder: $GOPATH/pkg/mod
+    go_build_cache: &go_build_cache
```

Comment:
This is a very big deal for `cross_task`. I would have expected this to also make a big difference for `validate_task`, but it doesn’t?!

A hypothesis to examine: would using a separate cache for the two tasks help? The current test uses a `cross`-created cache for `validate`; maybe the two differ enough in environment (some extra build tags??) that the caches don’t match?

Reply:
Yes: a separate build cache brings `validate_task` down to 0:42 when nothing in the code changed (vs. 4:19 with no caching at all, 3:13 with lint caches, and 3:10 with module cache).

That is _definitely_ with doing.